sdcm/utils/common.py

```diff
@@ -1065,19 +1062,24 @@ def list_clusters(self) -> list:  # pylint: disable=no-self-use
 
     clusters = EksCleaner().list_clusters()
 
-    if 'NodeType' in tags_dict:
-        tags_dict = tags_dict.copy()
-        del tags_dict['NodeType']
-
     if tags_dict:
-        clusters = filter_gce_by_tags(tags_dict=tags_dict, instances=clusters)
+        clusters = filter_k8s_clusters_by_tags(tags_dict, clusters)
 
     if verbose:
         LOGGER.info("Done. Found total of %s GKE clusters.", len(clusters))
 
     return clusters
 
 
+def filter_k8s_clusters_by_tags(tags_dict: dict,
+                                clusters: list[Union["EksCluster", "GkeCluster"]]) -> list[Union["EksCluster", "GkeCluster"]]:
```

Comment:
Why only EKS and GKE? We also support "kind" and "minikube" backends.

Reply:
> Inline comments plus following ones:
> 
> * There is no need to split tied logic into 3 different commits. combine "post_behavior" logic into one commit and leave "pylint" fixes in another one. So, you will have just 2 commits.

Yes, it is just a draft. i plan to merge into commit. i use just as steps.

> * Need to remove influence of "post_behavior" options which refer to db cluster and loader nodes.
>   Now, if you define any of those to "destroy" it will touch K8S cluster nodes.

we have post_behavior always enabled in test_default.yaml. So they always are loaded into configuratoin. But anyway, the post_behavior_db and post_behavior_loader are not affect on cleaning gke/eks when clean_resources running with --post_behavior

Reply:
But as i can see according code, we don't support such in clean_cloud_resources.
Could it be that kind and minikube will be cleared by clean_resources_docker?